### DISCUSSION TITLE
.NET MAUI 8.0.40 (Service Release 5)

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#22409] [Announcements]
This release is now available via [NuGet.org](https://www.nuget.org/packages/Microsoft.Maui.Controls/8.0.40) and is compatible with the latest .NET 8 workload. 

```
<ItemGroup>
  <PackageReference Include="Microsoft.Maui.Controls" Version="8.0.40" />
  <PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="8.0.40" />
</ItemGroup>
```

The focus of our service releases is resolving high-impact bugs for developers upgrading their applications from Xamarin to .NET 8.

- [Release notes](https://github.com/dotnet/maui/releases/tag/8.0.40)

**Comment by nll** [comment]
What is the _"latest .NET 8 workload"_ ? How do we make sure that we have it / are using it ?

**Reply by dartasen** [reply]
dotnet workload list in a cmd

**Reply by cytoph** [reply]
Is it just me or is the 8.0.40 workload not yet available? I see it on NuGet, but when I use `dotnet workload install` the version I get is still 8.0.21.

**Reply by dartasen** [reply]
Yes 8.0.21 workload is the latest workload yet.
Nugets are released independently from workloads that are shipped sometimes with visual studio release

**Reply by cytoph** [reply]
Aaah, that makes sense. Thank you for your fast answer. Is there any way to be notified if there's a newer version of the workload available? For releases here I could for example use the RSS feed and get notification when there's a new release, but I wouldn't know how to do the same for new workload versions.

**Reply by dartasen** [reply]
@cytoph Hmmm, i'm not aware of any easy pre-made solution but dotnet workload list is fetching some manifests to notify any update so I guess you could hit the api that exposes workload manifests and compare with your installed version

**Reply by cytoph** [reply]
Yeah, that was my idea, too. But I wasn't able to find out which api/endpoint it is calling for that 🤔

**Comment by eqmarcin** [comment]
Thanks. Hopefully fixes for memory leaks will be working as expected.

**Comment by xyeie** [comment]
The current version still experiences numerous performance issues; when can we expect a smooth user experience? Navigating to pages featuring CollectionView or BindingLayout is notably sluggish, with more complex pages suffering from even greater delays, to the point where it feels slower than web-based interfaces.

**Reply by MartyIX** [reply]
Not sure what platforms you are interested. However, web-based interfaces are imho much faster than what MAUI can achieve now on Windows because interop calls are simply too costly and MAUI does a lot of unnecessary calls right now. I suspect that it's somewhat similar for other supported platforms.

If there is a particular scenario you would want to optimize, you can profile your app and contribute a fix. That's what I have been doing https://github.com/dotnet/maui/pulls?q=is%3Apr+author%3AMartyIX+is%3Aclosed. It seems the fastest option at this point to optimize stuff one cares for. There are quite lot of "low hanging fruit" optimization opportunities. Some people are offendee by this suggestion but I see great value in any form of contribution (testing, debugging an issue, measuring perf, providing a PR). It counts for you and for the rest of us.

**Comment by GUELIANEBelkacem** [comment]
I'm happy that memory leaks are getting some attention, the most critical problem in my opinion, thank you and waiting for more fixes with SR6!

**Reply by thanigaiharan** [reply]
When  the .NET SR6 will be released?

**Reply by GUELIANEBelkacem** [reply]
You can check out the progress here:
https://github.com/orgs/dotnet/projects/194/views/31

All the SR6 tickets have been marked as finished, so we're getting it this week most likely.

**Reply by davidortinau** [reply]
SR6 is going through a final round of testing and validation. As soon as we have approval it'll go up to NuGet and I'll announce it.

**Reply by MartyIX** [reply]
![are we there yet?](https://y.yarn.co/7b03abfe-3ce0-41a2-bfd6-d40dd04f2eec_text.gif)

😄
